app/src/main/kotlin/com/github/premnirmal/ticker/model/IStocksProvider.kt

```diff
@@ -55,4 +55,8 @@ interface IStocksProvider {
     ticker: String,
     holding: Holding
   )
+
+  fun totalHoldings(): Pair<String, Int>
+
+  fun totalGainLoss(): Pair<String, String>
```

Comment:
Move this logic to a viewmodel and use the VM inside `HomeFragment`.

Prefer not to have calculation logic like this in the model here, instead if should be part of the UI binding code.

Reply:
Thanks @premnirmal for your feedback. My first thought for the totals were to use a dedicated stock quote block that displays the totals instead of the individual stock data, but this would require quite a large change in the design. Having the totals in the header looked as the perfect place when I created a mockup. Courious to see what else could be used!

The StocksProvider seemed like a perfect place for the totals calculation. Will check out the VM.